udp-native/src/main/java/org/ballerinalang/stdlib/udp/endpoint/ConnectionlessClientActions.java

```diff
@@ -123,42 +98,29 @@ public static Object initEndpoint(Environment env, BObject client, Object addres
         return null;
     }
 
-    private static String getHostFromAddress(BMap<BString, Object> addressRecord) {
-        String host = null;
-        try {
-            BString bHost = addressRecord.getStringValue(StringUtils.fromString(SocketConstants.CONFIG_FIELD_HOST));
-            if (bHost != null) {
-                host = bHost.getValue();
-            }
-        } catch (NullPointerException e) {
-            return null;
-        }
-        return host;
-    }
-
-    public static Object receiveFrom(Environment env, BObject client, long length) {
+    public static Object receive(Environment env, BObject client) {
         final Future balFuture = env.markAsync();
-        if (length != DEFAULT_EXPECTED_READ_LENGTH && length < 1) {
-            String msg = "requested byte length need to be 1 or more";
-            balFuture.complete(SocketUtils.createSocketError(msg));
-            return null;
-        }
         DatagramChannel socket = (DatagramChannel) client.getNativeData(SocketConstants.SOCKET_KEY);
         int socketHash = socket.hashCode();
         SocketService socketService = (SocketService) client.getNativeData(SocketConstants.SOCKET_SERVICE);
-        ReadPendingCallback readPendingCallback = new ReadPendingCallback(balFuture, (int) length, socketHash,
-                socketService.getReadTimeout());
+        ReadPendingCallback readPendingCallback = new ReadPendingCallback(balFuture, DEFAULT_EXPECTED_READ_LENGTH,
+                socketHash, socketService.getReadTimeout());
         ReadPendingSocketMap.getInstance().add(socket.hashCode(), readPendingCallback);
         log.debug("Notify to invokeRead");
         SelectorManager.getInstance().invokeRead(socketHash);
         return null;
     }
 
-    public static Object sendTo(BObject client, BArray content, BMap<BString, Object> address) {
+    public static Object send(BObject client, BMap<BString, Object> datagram) {
         DatagramChannel socket = (DatagramChannel) client.getNativeData(SocketConstants.SOCKET_KEY);
-        String host = address.getStringValue(StringUtils.fromString(SocketConstants.CONFIG_FIELD_HOST)).getValue();
-        int port = address.getIntValue(StringUtils.fromString(SocketConstants.CONFIG_FIELD_PORT)).intValue();
-        byte[] byteContent = content.getBytes();
+        BMap<BString, Object> remoteAddress = (BMap<BString, Object>) datagram
+                .getMapValue(StringUtils.fromString("remoteAddress"));
```

Comment:
Shall we make `remoteAddress` a string constant as same as `SocketConstants.CONFIG_FIELD_HOST`

Reply:
"remoteAddress" replaced with SocketConstants.DATAGRAM_REMOTE_ADDRESS